src/callback_handler/sigstore_verification.rs

```diff
@@ -16,8 +25,60 @@ impl Client {
         sources: Option<Sources>,
         fulcio_and_rekor_data: Option<&FulcioAndRekorData>,
     ) -> Result<Self> {
-        let verifier = Verifier::new(sources, fulcio_and_rekor_data)?;
-        Ok(Client { verifier })
+        let cosign_client = Arc::new(Mutex::new(Self::build_cosign_client(
```

Comment:
just a question. I see the usefulness of `build_cosign_client()`, but why do we need the cosign_client thread safe? I suppose I'm missing something obvious.

Reply:
It's not that obvious, that's not a silly question! :)

It has to be thread safe because it's a mutable object, that is shared via pointers inside between the `Verifier` and the `Client`. Rust borrow checker rightfully complained about that. That was the only way to allow the sharing between the two objects